model/task/task.go

```diff
@@ -464,11 +463,17 @@ func (t *Task) DependenciesMet(depCaches map[string]Task) (bool, error) {
 		if !t.SatisfiesDependency(&depTask) {
 			return false, nil
 		}
-		if depTask.FinishTime.After(latestTime) {
-			latestTime = depTask.FinishTime
-		}
 	}
-	t.DependenciesMetTime = latestTime
+	// this is not exact, but depTask.FinishTime is not always set in time to use that
+	t.DependenciesMetTime = time.Now()
+	err = UpdateOne(
+		bson.M{IdKey: t.Id},
+		bson.M{
+			"$set": bson.M{DependenciesMetTimeKey: t.DependenciesMetTime},
+		})
+	if err != nil {
+		return true, errors.Wrapf(err, "task.DependenciesMet() failed to update task '%s'", t.Id)
```

Comment:
Even though we return true with the error here, in the cases we use this function, we ignore the boolean value if there's an error. I think it might be better to just log the error, and still return true regardless, so that things can continue.